VMBackup/main/WaagentLib.py

```diff
@@ -2255,13 +2255,23 @@ def RunGetOutput(cmd, chk_err=True, log_cmd=True):
         LogIfVerbose(cmd)
     try:
         output = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
+        if isinstance(output, bytes):
+            output = output.decode('latin-1')
     except subprocess.CalledProcessError as e:
         if chk_err and log_cmd:
             Error('CalledProcessError.  Error Code is ' + str(e.returncode))
             Error('CalledProcessError.  Command string was ' + e.cmd)
-            Error('CalledProcessError.  Command result was ' + (e.output[:-1]).decode('latin-1'))
-        return e.returncode, e.output.decode('latin-1')
-    return 0, output.decode('latin-1')
+            if isinstance(e.output[:-1], bytes):
+                Error('CalledProcessError.  Command result was ' + (e.output[:-1]).decode('latin-1'))
```

Comment:
my earlier question was, if we check in the try block itself, do we even need to check twice in the except block.

Reply:
Ya, like if there is an exception raised even before checking for this condition then we would be needing them.